en/micro-integrator/docs/develop/creating-artifacts/creating-an-api.md

```diff
@@ -1,38 +1,171 @@
+            </td>
+        </tr>
+        <tr>
+            <th>
+                Import API Artifact
+            </th>
+            <td>
+                Select this option to import an existing REST API configuration (XML definition) that was created using WSO2 Integration Studio.
+            </td>
+        </tr>
+    </table>
+
+3.  Click **Next** to go to the next page and enter the relevant details.
+    
+    -   If you selected **Create a New API** in the previous step, enter the basic details that are required for creating the synapse configuration (XML) of the API:
+
+        <img src="../../../assets/img/create_artifacts/new_api/new-api-artifact-dialog.png" width="500">
+
+        <table>
+            <tr>
+                <th>
+                    Parameter
+                </th>
+                <th>
+                    Description
+                </th>
+            </tr>
+            <tr>
+                <td>
+                    Name
+                </td>
+                <td>
+                    <b>Required</b>.</br>
+                    The name of the REST API.
+                </td>
+            </tr>
+            <tr>
+                <td>
+                    Context
+                </td>
+                <td>
+                    <b>Required</b>.</br>
+                    The context for the REST API. For example, <code>/healthcare</code>.
+                </td>
+            </tr>
+            <tr>
+                <td>
+                    Path to Swagger Definition
+                </td>
+                <td>
+                    Enter the path to a custom Swagger definition (YAML file) that is stored in a registry project in your workspace.</br></br>
+                    Once this API is created and deployed in the Micro Integrator, users will be able to access this custom Swagger definition in addition to the default Swagger definition of the API.
+                </td>
+            </tr>
+        </table>
+
+    -   If you selected **Generate API using Swagger Definition** in the previous step, enter the details of your custom Swagger file:
+
+        <img src="../../../assets/img/create_artifacts/new_api/new-api-swagger-dialog.png" width="500">
+        
+        <table>
+            <tr>
+                <th>
+                    Parameter
+                </th>
+                <th>
+                    Description
+                </th>
+            </tr>
+            <tr>
+                <td>
+                    Swagger Definition File
+                </td>
+                <td>
+                    <b>Required</b>.</br>
+                    Click <b>Browse</b> and select the Swagger file.
+                </td>
+            </tr>
+            <tr>
+                <td>
+                    Swagger Registry Path
+                </td>
+                <td>
+                    <b>Required</b>.</br></br>
```

Comment:
This is not a Required parameter. If user skip this, then the original swagger file will not be reflected when we call the API with ?swagger.json queryparam